apps/collection/js/setup.js

```diff
@@ -32,28 +32,16 @@
 
       this.initializing = true;
 
-      var xhr = new XMLHttpRequest();
-      xhr.overrideMimeType('application/json');
-      xhr.open('GET', PRE_INSTALLED_COLLECTIONS_FILE, true);
-
-      xhr.onload = function _xhrOnLoadFile() {
-        if (!(xhr.status === 200 | xhr.status === 0)) {
-          this.onError('Unknown response when getting data.');
-          return;
-        }
-
-        try {
-          this.populate(JSON.parse(xhr.responseText));
-        } catch (ex) {
-          this.onError(ex);
-        }
-      }.bind(this);
-
-      xhr.onerror = function _xhrOnError() {
-        this.onError('file not found');
-      }.bind(this);
-
-      xhr.send();
+      LazyLoader.getJSON(PRE_INSTALLED_COLLECTIONS_FILE)
+        .then(function(json) {
+          try {
+            this.populate(JSON.parse(json));
+          } catch (ex) {
+            this.onError(ex);
+          }
```

Comment:
As mentioned before this won't work as the function isn't bound to the correct `this` reference. Look at the existing code and what it does.